src/main/java/com/commercetools/sync/taxcategories/utils/TaxRatesUpdateActionUtils.java

```diff
@@ -116,29 +124,25 @@ private TaxRatesUpdateActionUtils() {
     private static List<UpdateAction<TaxCategory>> buildRemoveOrReplaceTaxRateUpdateActions(
         @Nonnull final List<TaxRate> oldTaxRates,
         @Nonnull final List<TaxRateDraft> newTaxRatesDrafts) {
-        final Map<String, TaxRateDraft> taxRateDraftCountryStateMap = newTaxRatesDrafts
-                .stream()
-                .collect(
-                    toMap(taxRateDraft -> taxRateDraft.getCountry() + "_" + taxRateDraft.getState(),
-                        taxRateDraft -> taxRateDraft));
 
         return oldTaxRates
             .stream()
-            .map(oldTaxRate -> {
-                final CountryCode oldTaxRateCountryCode = oldTaxRate.getCountry();
-                final String oldTaxRateState = oldTaxRate.getState();
-                final TaxRateDraft matchingNewTaxRateDraft = taxRateDraftCountryStateMap
-                        .get(oldTaxRateCountryCode + "_" + oldTaxRateState);
-                return ofNullable(matchingNewTaxRateDraft)
-                    .map(taxRateDraft -> {
-                        if (!hasSameFields(oldTaxRate, taxRateDraft)) {
-                            return singletonList(ReplaceTaxRate.of(oldTaxRate.getId(), taxRateDraft));
-                        } else {
-                            return new ArrayList<UpdateAction<TaxCategory>>();
-                        }
-                    })
-                    .orElseGet(() -> singletonList(RemoveTaxRate.of(oldTaxRate.getId())));
-            })
+            .map(oldTaxRate -> newTaxRatesDrafts
+                .stream()
+                .filter(taxRateDraft -> Objects.equals(oldTaxRate.getCountry(), taxRateDraft.getCountry()))
+                .filter(taxRateDraft -> oldTaxRate.getState() == null
+                    || (oldTaxRate.getState() != null && taxRateDraft.getState() == null)
```

Comment:
Here why we include the taxRateDraft when either new state or old state is null?

Reply:
When you unset state in the draft. This is also could be considered as a ReplaceAction rather than using Removing and Adding the same tax rate.

check the scenario: 
buildTaxRatesUpdateActions_WithRemovedState_ShouldReturnOnlyReplaceAction